src/governance/timelock/utils/operation_state.cairo

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.14.0 (governance/timelock/utils/operation_state.cairo)
+
+use core::fmt::{Debug, Formatter, Error};
+
+#[derive(Drop, Copy, Serde, PartialEq)]
+pub enum OperationState {
+    Unset,
+    Waiting,
+    Ready,
+    Done
+}
+
+impl DebugOperationState of core::fmt::Debug<OperationState> {
```

Comment:
> Do we need the Debug implementation outside tests? If not, I think we should move it into the common.cairo file there

Users might if they're testing their contract's implementation of timelock, no?—I suppose they can use the other getters, but IMO it seems unnecessarily restrictive

> And then we probably don't need this operation_state file, since the enum can be defined in the component itself (not extra implementations).

I considered defining the enum in the component. The interface, however, needs the enum in scope for `get_operation_state`, so this would create a circular dependency between `interface` and `timelock_controller`. An alternative would be to define the enum in `interface`. This doesn't seem right to me though. If we really don't want this module, we can maybe define it directly in the utils mod

Reply:
Why not directly derive Debug in the enum?

Reply:
> I considered defining the enum in the component. The interface, however, needs the enum in scope for get_operation_state, so this would create a circular dependency between interface and timelock_controller

It should compile anyway, right?

Reply:
> Why not directly derive Debug in the enum?

Ahh good call

> It should compile anyway, right?

It will, yeah. I just think the dep flow is weird with interface. I'll update and see what we think